### ISSUE TITLE
New streaming code does not send `byebye` event

### ISSUE THREAD

**Issue by bartekn** [stellar/go#1934]
New streaming code does not send `byebye` event before closing a connection. Compare:
```
curl -H "Accept: text/event-stream" "https://horizon-stg-pubnet.stellar.org/order_book?selling_asset_type=credit_alphanum4&selling_asset_code=BB1&selling_asset_issuer=GD5J6HLF5666X4AZLTFTXLY46J5SW7EXRKBLEYPJP33S33MXZGV6CWFN&buying_asset_type=native&limit=30&c=0.5129643434554865"
```
and
```
curl -H "Accept: text/event-stream" "https://horizon.stellar.org/order_book?selling_asset_type=credit_alphanum4&selling_asset_code=BB1&selling_asset_issuer=GD5J6HLF5666X4AZLTFTXLY46J5SW7EXRKBLEYPJP33S33MXZGV6CWFN&buying_asset_type=native&limit=30&c=0.5129643434554865"
```
The later sends:
```
retry: 10
event: close
data: "byebye"
```
before closing the connection that is sent via `services/horizon/internal/render/sse/Stream.Done()` function. I think it's not called from `StreamHandler`. Introduced in 0.21.0. Would be good to fix in 0.23.0, if not then 0.24.0.

Git blame: https://github.com/stellar/go/blame/04a852a4d23fe04e07dd70e96331594b30c60cbd/services/horizon/internal/render/sse/stream_handler.go#L76

**Comment by bartekn** [comment]
Moving to 0.24.0.

**Comment by bartekn** [comment]
Closed in #1942.
